tools/Tasks.ps1

```diff
@@ -0,0 +1,200 @@
+    [Environment]::GetEnvironmentVariable($envVariableName)
+    Write-Host ""
+    
+    Write-Host "If your needs demand you to set connection string differently, for all the supported Key/Value pairs in connection string"
+    #Write-Host "`$env:$envVariableName=SubscriptionId=<subid>;AADTenantId=<tenantId>"
+    Write-Host "Please visit https://github.com/Azure/azure-powershell/blob/dev/documentation/Using-Azure-TestFramework.md"
+}
+
+Function List-BuildScopes()
+{
+    Write-Host "Below are available scopes you can specify for building specific projects"
+    Write-Host ""    
+    Get-ChildItem -path "$env:repoRoot\src\ResourceManager" -dir | Format-Wide -Column 5 | Format-Table -Property Name    
+    Get-ChildItem -path "$env:repoRoot\src\ServiceManagement" -dir | Format-Wide -Column 5 | Format-Table -Property Name
+}
+
+Function Build-Repo([string] $scope)
+{
+    Write-Host "Executing Build-Repo"    
+    
+    if([string]::IsNullOrEmpty($scope) -eq $true)
+    {
+       Write-Host "Starting Full build"
+       msbuild.exe "$env:repoRoot\build.proj" /t:Build
+    }
+    else
+    {
+        Write-Host "cmdline Args: msbuild.exe $env:repoRoot\build.proj /p:Scope=$scope"
+        msbuild.exe "$env:repoRoot\build.proj" /t:Build /p:Scope=$scope
+    }
+}
+
+Function Run-CheckinTests()
+{
+    Write-Host "cmdline Args: msbuild.exe $env:repoRoot\build.proj /t:Test"
+    msbuild.exe "$env:repoRoot\build.proj" /t:Test
+}
+
+Function Run-MockedScenarioTests()
+{
+    Write-Host "cmdline Args: msbuild.exe $env:repoRoot\build.proj /t:Test"
+    msbuild.exe "$env:repoRoot\build.proj" /t:"Build;BeforeRunTests;MockedScenarioTests"
+}
+
+Function Create-ServicePrincipal-Help()
+{
+    #TODO: Find a way to get service principal secret key during SPN creation
+    #TODO: Validate user input
+    Write-Host "Visit https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-service-principal-portal/ to know more about how to create service principal manually"
+
+    #Add-AzureRmAccount
+
+    #$appName = (Read-Host "Provide an app name (without spaces) for the new AD App that will be created").Trim()
+    #$tenantName = (Read-Host "Provide your tenant name (e.g. if you are using Microsoft tenant, tenant Name will be microsoft.com").Trim()
+    #$pwd = (Read-Host "Provide a password for the new AD app that will be created").Trim()
+
+    #$appUri = [string]::Format("https://{0}/{1}", $tenantName, $appName)
+
+    #$app = AzureRmADApplication -DisplayName $appName -HomePage $appUri -IdentifierUris $appUri -Password $pwd
+
+    #New-AzureRmADServicePrincipal -ApplicationId $app.ApplicationId
+    #New-AzureRmRoleAssignment -RoleDefinitionName Reader -ServicePrincipalName $app.ApplicationId.Guid
+}
+
+Function Print-ConnectionString([string]$subId, [string]$aadTenant, [string]$spn, [string]$spnSecret, [string]$recordMode, [string]$uris)
```

Comment:
Should we have it actually set the connection string too?  Thinking that setting the connection string and launching VS is what most people are goint to want to do.  This functionbality for priting the connection string is definitely necessary, however.